src/libraries/System.Text.Json/src/System/ReflectionExtensions.cs

```diff
@@ -43,20 +43,20 @@ public static bool IsAssignableFromInternal(this Type type, Type from)
         private static bool HasJsonConstructorAttribute(ConstructorInfo constructorInfo)
             => constructorInfo.GetCustomAttribute<JsonConstructorAttribute>() != null;
 
-        public static bool HasRequiredMemberAttribute(this ICustomAttributeProvider memberInfo)
+        public static bool HasRequiredMemberAttribute(this MemberInfo memberInfo)
         {
             // For compiler related attributes we should only look at full type name rather than trying to do something different for version when attribute was introduced.
             // I.e. library is targeting netstandard2.0 with polyfilled attributes and is being consumed by an app targeting net7.0 or greater.
-            return memberInfo.HasCustomAttributeWithName("System.Runtime.CompilerServices.RequiredMemberAttribute", inherit: true);
+            return memberInfo.HasCustomAttributeWithName("System.Runtime.CompilerServices.RequiredMemberAttribute", inherit: false);
```

Comment:
>false [](http://example.com/codeflow?start=125&length=5)

does the perf gain mostly coming from this change and the value type check in the other file? I am not seeing any other change that can affect the perf.